File: mobile_apps/ALADDIN_iOS/Components/Modals/RecoveryOptionsModal.swift
import SwiftUI

/**
 * 🔑 Recovery Options Modal
 * Модальное окно выбора способа восстановления доступа
 * Окно #6 - показывает 4 способа восстановления
 */

struct RecoveryOptionsModal: View {
    
    @Binding var isPresented: Bool
    
    @State private var showQRScanner: Bool = false
    @State private var showManualInput: Bool = false
    @State private var scannerMode: QRScannerModal.ScanMode = .recoveryQR
    
    var onRecoveryComplete: () -> Void
    
    // MARK: - Body
    
    var body: some View {
        ZStack {
            // Backdrop blur
            Color.black.opacity(0.5)
                .ignoresSafeArea()
                .blur(radius: 20)
            
            // Modal content
            VStack(spacing: Spacing.xl) {
                // Header
                VStack(spacing: Spacing.m) {
                    Text("🔑")
                        .font(.system(size: 40))
                    
                    Text("ВОССТАНОВИТЬ ДОСТУП")
                        .font(.system(size: 20, weight: .bold))
                        .foregroundColor(.secondaryGold)
                    
                    Text("Как вы хотите восстановить?")
                        .font(.system(size: 14))
                        .foregroundColor(.textSecondary)
                }
                
                // Recovery options
                VStack(spacing: Spacing.m) {
                    // Option 1: Through family member (BEST!)
                    RecoveryOptionButton(
                        icon: "👨‍👩‍👧‍👦",
                        title: "ЧЕРЕЗ ЧЛЕНА СЕМЬИ",
                        subtitle: "(если у кого-то есть доступ)",
                        color: .successGreen
                    ) {
                        scannerMode = .recoveryFromFamily
                        showQRScanner = true
                    }
                    
                    // Option 2: Scan saved QR #2
                    RecoveryOptionButton(
                        icon: "📷",
                        title: "СКАНИРОВАТЬ QR #2",
                        subtitle: "(сохранённый код восстановления)",
                        color: .primaryBlue
                    ) {
                        scannerMode = .recoveryQR
                        showQRScanner = true
                    }
                    
                    // Option 3: Enter code manually
                    RecoveryOptionButton(
                        icon: "🔤",
                        title: "ВВЕСТИ КОД ВРУЧНУЮ",
                        subtitle: "(FAM-A1B2-C3D4-E5F6)",
                        color: .secondaryGold
                    ) {
                        showManualInput = true
                    }
                    
                    // Option 4: Contact support
                    RecoveryOptionButton(
                        icon: "📧",
                        title: "ОБРАТИТЬСЯ В ПОДДЕРЖКУ",
                        subtitle: "(если всё потеряно)",
                        color: .dangerRed
                    ) {
                        contactSupport()
                    }
                }
                
                // Back button
                Button(action: { isPresented = false }) {
                    Text("НАЗАД")
                        .font(.body)
                        .foregroundColor(.textSecondary)
                }
            }
            .padding(Spacing.xl)
            .frame(width: 340)
            .background(
                LinearGradient(
                    colors: [Color(hex: "#1e3a5f"), Color(hex: "#2e5090")],
                    startPoint: .topLeading,
                    endPoint: .bottomTrailing
                )
            )
            .cornerRadius(24)
            .shadow(color: .black.opacity(0.5), radius: 30, x: 0, y: 20)
        }
        .sheet(isPresented: $showQRScanner) {
            QRScannerModal(
                isPresented: $showQRScanner,
                mode: scannerMode,
                onCodeScanned: handleCodeScanned
            )
        }
        .sheet(isPresented: $showManualInput) {
            ManualCodeInputModal(
                isPresented: $showManualInput,
                onCodeEntered: handleCodeScanned
            )
        }
    }
    
    // MARK: - Actions
    
    private func handleCodeScanned(_ code: String) {
        // TODO: Validate and restore family
        print("✅ Code scanned: \(code)")
        isPresented = false
        onRecoveryComplete()
    }
    
    private func contactSupport() {
        if let url = URL(string: "mailto:support@aladdin.family?subject=Восстановление доступа к семье") {
            UIApplication.shared.open(url)
        }
    }
}

// MARK: - Recovery Option Button

struct RecoveryOptionButton: View {
    let icon: String
    let title: String
    let subtitle: String
    let color: Color
    let action: () -> Void
    
    var body: some View {
        Button(action: action) {
            HStack(spacing: Spacing.m) {
                // Icon
                Text(icon)
                    .font(.system(size: 32))
                
                // Text
                VStack(alignment: .leading, spacing: 4) {
                    Text(title)
                        .font(.system(size: 15, weight: .bold))
                        .foregroundColor(.white)
                    
                    Text(subtitle)
                        .font(.system(size: 12))
                        .foregroundColor(.textSecondary)
                }
                
                Spacer()
                
                // Arrow
                Image(systemName: "chevron.right")
                    .font(.system(size: 14, weight: .semibold))
                    .foregroundColor(.textSecondary)
            }
            .frame(height: 72)
            .padding(.horizontal, Spacing.m)
            .background(color.opacity(0.15))
            .cornerRadius(12)
            .overlay(
                RoundedRectangle(cornerRadius: 12)
                    .stroke(color, lineWidth: 1)
            )
        }
        .buttonStyle(.plain)
    }
}

// MARK: - Preview

struct RecoveryOptionsModal_Previews: PreviewProvider {
    static var previews: some View {
        RecoveryOptionsModal(
            isPresented: .constant(true),
            onRecoveryComplete: {}
        )
    }
}

File: mobile_apps/ALADDIN_Android_New/app/src/main/java/family/aladdin/android/ui/components/modals/RecoveryOptionsModal.kt
package family.aladdin.android.ui.components.modals

import android.content.Intent
import android.net.Uri
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.window.Dialog
import androidx.compose.ui.window.DialogProperties
import family.aladdin.android.ui.theme.*

/**
 * 🔑 Recovery Options Modal
 * Модальное окно выбора способа восстановления доступа
 * Окно #6 - показывает 4 способа восстановления
 * 
 * РАЗМЕРЫ (проверено):
 * - Modal width: 340dp
 * - Modal height: 520dp (безопасно!)
 * - Each button: 72dp height
 * - Total content: ~500dp
 */

@Composable
fun RecoveryOptionsModal(
    onRecoveryComplete: () -> Unit,
    onDismiss: () -> Unit
) {
    val context = LocalContext.current
    var showQRScanner by remember { mutableStateOf(false) }
    var showManualInput by remember { mutableStateOf(false) }
    var scannerMode by remember { mutableStateOf(QRScanMode.RECOVERY_QR) }
    
    Dialog(
        onDismissRequest = onDismiss,
        properties = DialogProperties(
            dismissOnBackPress = true,
            dismissOnClickOutside = false
        )
    ) {
        Card(
            modifier = Modifier
                .width(340.dp)
                .wrapContentHeight(),
            shape = RoundedCornerShape(24.dp),
            colors = CardDefaults.cardColors(
                containerColor = Color.Transparent
            )
        ) {
            Box(
                modifier = Modifier
                    .background(
                        Brush.linearGradient(
                            colors = listOf(
                                Color(0xFF0F172A),  // Космический тёмный
                                Color(0xFF1E3A8A),  // Глубокий синий
                                Color(0xFF3B82F6),  // Электрический синий
                                Color(0xFF1E40AF)   // Королевский синий
                            )
                        )
                    )
                    .padding(24.dp)
            ) {
                Column(
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.spacedBy(SpacingXL)
                ) {
                    // Header
                    Column(
                        horizontalAlignment = Alignment.CenterHorizontally,
                        verticalArrangement = Arrangement.spacedBy(SpacingM)
                    ) {
                        Text(
                            text = "🔑",
                            fontSize = 40.sp
                        )
                        
                        Text(
                            text = "ВОССТАНОВИТЬ ДОСТУП",
                            fontSize = 20.sp,
                            fontWeight = FontWeight.Bold,
                            color = Color(0xFFFCD34D)  // Яркое золото!
                        )
                        
                        Text(
                            text = "Как вы хотите восстановить?",
                            fontSize = 14.sp,
                            color = TextSecondary
                        )
                    }
                    
                    // Recovery options (4 кнопки)
                    Column(verticalArrangement = Arrangement.spacedBy(SpacingM)) {
                        // Option 1: Through family member (BEST!)
                        RecoveryOptionButton(
                            icon = "👨‍👩‍👧‍👦",
                            title = "ЧЕРЕЗ ЧЛЕНА СЕМЬИ",
                            subtitle = "(если у кого-то есть доступ)",
                            color = Color(0xFF10B981),  // Зелёный
                            onClick = {
                                scannerMode = QRScanMode.RECOVERY_FROM_FAMILY
                                showQRScanner = true
                            }
                        )
                        
                        // Option 2: Scan saved QR #2
                        RecoveryOptionButton(
                            icon = "📷",
                            title = "СКАНИРОВАТЬ QR #2",
                            subtitle = "(сохранённый код восстановления)",
                            color = Color(0xFF60A5FA),  // Электрический синий из иконки!
                            onClick = {
                                scannerMode = QRScanMode.RECOVERY_QR
                                showQRScanner = true
                            }
                        )
                        
                        // Option 3: Enter code manually
                        RecoveryOptionButton(
                            icon = "🔤",
                            title = "ВВЕСТИ КОД ВРУЧНУЮ",
                            subtitle = "(FAM-A1B2-C3D4-E5F6)",
                            color = Color(0xFFFCD34D),  // Яркое золото из иконки!
                            onClick = {
                                showManualInput = true
                            }
                        )
                        
                        // Option 4: Contact support
                        RecoveryOptionButton(
                            icon = "📧",
                            title = "ОБРАТИТЬСЯ В ПОДДЕРЖКУ",
                            subtitle = "(если всё потеряно)",
                            color = Color(0xFFEF4444),  // Красный
                            onClick = {
                                val intent = Intent(Intent.ACTION_SENDTO).apply {
                                    data = Uri.parse("mailto:support@aladdin.family?subject=Восстановление доступа к семье")
                                }
                                context.startActivity(intent)
                            }
                        )
                    }
                    
                    // Back button
                    Text(
                        text = "НАЗАД",
                        fontSize = 14.sp,
                        color = TextSecondary,
                        modifier = Modifier.clickable(onClick = onDismiss)
                    )
                }
            }
        }
    }
    
    if (showQRScanner) {
        QRScannerModal(
            mode = scannerMode,
            onCodeScanned = { _ ->
                showQRScanner = false
                onRecoveryComplete()
            },
            onDismiss = { showQRScanner = false }
        )
    }
    
    if (showManualInput) {
        ManualCodeInputModal(
            onCodeEntered = { _ ->
                showManualInput = false
                onRecoveryComplete()
            },
            onDismiss = { showManualInput = false }
        )
    }
}

@Composable
fun RecoveryOptionButton(
    icon: String,
    title: String,
    subtitle: String,
    color: Color,
    onClick: () -> Unit
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .height(72.dp)  // Фиксированная высота
            .background(
                color.copy(alpha = 0.15f),
                RoundedCornerShape(12.dp)
            )
            .border(
                width = 1.dp,
                color = color,
                shape = RoundedCornerShape(12.dp)
            )
            .clickable(onClick = onClick)
            .padding(horizontal = SpacingM),
        horizontalArrangement = Arrangement.spacedBy(SpacingM),
        verticalAlignment = Alignment.CenterVertically
    ) {
        // Icon
        Text(
            text = icon,
            fontSize = 32.sp
        )
        
        // Text
        Column(
            verticalArrangement = Arrangement.spacedBy(4.dp),
            modifier = Modifier.weight(1f)
        ) {
            Text(
                text = title,
                fontSize = 15.sp,
                fontWeight = FontWeight.Bold,
                color = Color.White
            )
            
            Text(
                text = subtitle,
                fontSize = 12.sp,
                color = TextSecondary
            )
        }
        
        // Arrow
        Text(
            text = "›",
            fontSize = 24.sp,
            color = TextSecondary
        )
    }
}



